File: 238-product-of-array-except-self/product-of-array-except-self.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int mul = 1, c=0;
        vector<int> answer;
        c = count(nums.begin(), nums.end(), 0);
        if(c > 1) {
            for(int i=0; i<nums.size(); i++) {
                answer.push_back(0);
            }
        }
        else if (!c) {
            for(int i=0; i<nums.size(); i++) {
                mul = mul * nums[i];
            }
            
            for(int i=0; i<nums.size(); i++) {
                answer.push_back(int(mul/nums[i]));
            }
        }
        else {
            for(int i=0; i<nums.size(); i++) {
                if(nums[i] == 0) {
                    continue;
                }
                mul = mul * nums[i];
            }
            for(int i=0; i<nums.size(); i++) {
                if(nums[i] == 0) {
                    answer.push_back(mul);
                }
                else {
                    answer.push_back(0);
                }
            }
        }
        return answer;
    }
};

File: 238-product-of-array-except-self/product-of-array-except-self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:

        n = len(nums)
        mul = 1
        
        def arr(n):
            mul = 1
            print(n)
            for i in n:
                if i != 0:
                    mul *= i
            
            return mul

        
        if 0 in nums:
            d = Counter(nums)
            if d[0] > 1:
                return [0] * n
            elif d[0] == 1:
                x = nums.index(0)
                temp = arr(nums[:x])
                if x < n-1:
                    temp *= arr(nums[x+1:])
                
                result = [0] * n
                result[x] = temp
                return result
        else:
            mul = arr(nums)
            result = []
            for i in nums:
                result.append(mul//i)
            
        return result


        

            
        